dynamic-client-registration/dynamic-client-registration-mediator/src/main/java/org/wso2/financial/services/apim/mediation/policies/dynamic/client/registration/DynamicClientRegistrationRequestMediator.java

```diff
@@ -0,0 +1,337 @@
+
+import java.net.MalformedURLException;
+import java.text.ParseException;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.core.MediaType;
+
+/**
+ * Mediator for dynamic client registration.
+ */
+public class DynamicClientRegistrationRequestMediator extends AbstractMediator {
+
+    private static final Log log = LogFactory.getLog(DynamicClientRegistrationRequestMediator.class);
+
+    private boolean validateRequestJWT;
+    private String jwksEndpointName;
+    private String clientNameAttributeName;
+    private boolean useSoftwareIdAsAppName;
+    private int jwksEndpointTimeout;
+
+    @Override
+    public boolean mediate(MessageContext messageContext) {
+
+        org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) messageContext)
+                .getAxis2MessageContext();
+        Map<String, Object> headers = (Map<String, Object>)
+                axis2MessageContext.getProperty(org.apache.axis2.context.MessageContext.TRANSPORT_HEADERS);
+
+        String httpMethod = (String) messageContext.getProperty(DCRConstants.HTTP_METHOD);
+
+        // Check if the HTTP method is POST or PUT to process DCR JWT validation and request alteration
+        if (HttpMethod.POST.equals(httpMethod) || HttpMethod.PUT.equals(httpMethod)) {
+            try {
+                log.info("Processing DCR JWT validation and request alteration.");
+                String contentType = headers.containsKey(DCRConstants.CONTENT_TYPE_TAG) ?
+                        headers.get(DCRConstants.CONTENT_TYPE_TAG).toString() : null;
+                Optional<String> requestPayload = DCRUtil.buildMessagePayloadFromMessageContext(axis2MessageContext,
+                        contentType);
+                if (requestPayload.isPresent()) {
+                    if (Objects.requireNonNull(contentType).contains(DCRConstants.JWT_CONTENT_TYPE)) {
+                        // Perform JWT Validation and Payload conversion and modification to JWT payloads
+                        validateAndModifyPayload(messageContext, requestPayload.get(), headers);
+                    } else {
+                        // Perform Payload conversion and modification to JSON payloads
+                        modifyJsonPayload(messageContext, requestPayload.get(), headers);
+                    }
+                } else {
+                    log.error("Request payload is empty or not present.");
+                    DCRUtil.returnSynapseHandlerJSONError(messageContext, DCRConstants.BAD_REQUEST_CODE,
+                            DCRUtil.getErrorResponse(DCRConstants.INVALID_REQUEST,
+                                    "Request payload is empty or not present."));
+                    return false;
+                }
+
+            } catch (DCRHandlingException e) {
+                DCRUtil.returnSynapseHandlerJSONError(messageContext, e.getHttpCode(),
```

Comment:
shall we log e clause?

Reply:
Fixed in https://github.com/wso2/financial-services-apim-mediation-policies/pull/4/commits/ed9c9b371fe9fe98524452d688b02f7070d7fe08